workshops/create-custom-supply-chains.yaml

```diff
@@ -0,0 +1,294 @@
+  #! The value can be: beginner, intermediate, advanced, or extreme.
+  difficulty: intermediate
+  #! The duration field gives the maximum amount of time the workshop takes to complete. 
+  #! This field provides informational value and does not guarantee how long a workshop instance lasts. 
+  #! The field format is an integer number with s, m, or h suffix.
+  duration: 3h
+  #! url: https://github.com/mrgaryg/tap-cartographer-workshop
+  #! url: #@ data.values.workshop_git.git_url + "/" + data.values.workshop_git.git_org_name + "/" + data.values.workshop_git.git_repo
+  url: https://github.com/dkhopade/tap-cartographer-workshop
+  content:
+    image: #@ "{}/tap-workshop/cartographer-workshop:latest".format(data.values.container_registry.hostname)
+    files: #@ "{}/{}/tap-cartographer-workshop/workshop?ref=se-supplychains-workshop-dev".format(data.values.workshop_git.git_url,data.values.workshop_git.git_org_name)
+  session:
+    resources:
+      memory: 3Gi
+    namespaces:
+      budget: custom
+    applications:
+      terminal:
+        enabled: true
+        layout: split
+      console:
+        enabled: true
+        vendor: octant
+      editor:
+        enabled: true
+    dashboards:
+      - name: Cartographer Docs
+        url: https://cartographer.sh/docs/development/
+      - name: TAP GUI
+        url: #@ "http://tap-gui.{}/supply-chain/".format(data.values.tap.ingress_domain)
+      - name: Carto Live Editor
+        url: https://cartographer.sh/live-editor/index.htm
+    env: 
+    - name: TAP_INGRESS
+      value: #@ data.values.tap.ingress_domain
+    - name: TAP_CNRS_SUBDOMAIN
+      value: #@ data.values.tap.cnrs_subdomain
+    - name: CONTAINER_REGISTRY_HOSTNAME
+      value: #@ data.values.container_registry.hostname
+    - name: CONTAINER_REGISTRY_USERNAME
+      value: #@ data.values.container_registry.username
+    - name: CONTAINER_REGISTRY_PASSWORD
+      value: #@ data.values.container_registry.password
+    - name: CONTAINER_REGISTRY_REPOSITORY
+      value: #@ data.values.container_registry.repository
+    - name: GITOPS_REPOSITORY
+      value: #@ "{}/$(session_namespace).git".format(data.values.gitops.https_repository_prefix)
+    - name: GITOPS_REPOSITORY_USERNAME
+      value: #@ data.values.gitops.basic_auth_username
+    - name: GITOPS_REPOSITORY_PASSWORD
+      value: #@ data.values.gitops.basic_auth_password
+    objects:
+      - apiVersion: v1
+        kind: LimitRange
+        metadata:
+          name: resource-limits
+        spec:
+          limits:
+            - type: PersistentVolumeClaim
+              max:
+                storage: 10Gi
+      #! Permissions for tap-sa (Supply Chain)
+      - apiVersion: v1
+        kind: Secret
+        metadata:
+          name: tanzu-net-credentials
+        type: kubernetes.io/dockerconfigjson
+        data:
+          #@ docker_auth = base64.encode("{}:{}".format(data.values.tanzunet.username, data.values.tanzunet.password))
```

Comment:
We don't have a Tanzunet username and password available via the WaaS catalog currently, so we'd need to add that.

Reply:
So we found that we need this when we build an image; we can think of taking it off in future once we have a way within WaaS environment